django/sierra/sierra/tests/test_routers.py

```diff
@@ -0,0 +1,124 @@
+    """
+    Test that SierraRouter.db_for_{op} (db_for_read or db_for_write)
+    gives the expected result for the given model_class and database
+    """
+    test_method = getattr(routers.SierraRouter, 'db_for_{}'.format(op))
+    router, model = routers.SierraRouter(), model_class()
+    try:
+        result = test_method(router, model) == database
+    except Exception as e:
+        result = type(e)
+    assert result == expected
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('model_class,database,testing,expected', [
+    (EXPORT_TEST_MODEL_CLASS, 'default', False, True),
+    (EXPORT_TEST_MODEL_CLASS, 'default', True, True),
+    (EXPORT_TEST_MODEL_CLASS, 'sierra', False, False),
+    (EXPORT_TEST_MODEL_CLASS, 'sierra', True, False),
+    (SIERRA_TEST_MODEL_CLASS, 'sierra', False, False),
+    (SIERRA_TEST_MODEL_CLASS, 'sierra', True, True),
+    (SIERRA_TEST_MODEL_CLASS, 'default', False, False),
+    (SIERRA_TEST_MODEL_CLASS, 'default', True, False),
+])
+def test_sierra_router_allow_migrate(model_class, database, testing, expected):
+    """
+    Test that SierraRouter.allow_migrate returns the expected result
+    for the given model_class and database, depending on whether or not
+    it's a testing environment
+    """
+    router, model = routers.SierraRouter(), model_class()
+    settings.TESTING = testing
+    result = router.allow_migrate(database, model)
+    settings.TESTING = True
+    assert result == expected
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('model_class,database,expected', [
+    (EXPORT_TEST_MODEL_CLASS, 'default', True),
+    (SIERRA_TEST_MODEL_CLASS, 'sierra', True),
+    (EXPORT_TEST_MODEL_CLASS, 'sierra', False),
+    (SIERRA_TEST_MODEL_CLASS, 'default', False),
+])
+def test_table_in_database(model_class, database, expected):
+    """
+    Test that the given model_class's table is (or isn't) in the given
+    database
+    """
+    cursor = connections[database].cursor()
+    tables = cursor.db.introspection.django_table_names(only_existing=True)
+    assert (model_class._meta.db_table in tables) == expected
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('model_class,database', [
+    (EXPORT_TEST_MODEL_CLASS, 'default'),
+    (SIERRA_TEST_MODEL_CLASS, 'sierra'),
+])
+def test_db_read_routing(model_class, database):
+    """
+    Test that the given model_class actually reads from the given
+    database
+    """
+    queryset = model_class.objects.all()
+    assert queryset.db == database
+
+
+# The next test only tests that the EXPORT_TEST_MODEL_CLASS writes to
+# the default database. It doesn't test SIERRA_TEST_MODEL_CLASS because
+# the save method on the base.models.ReadOnly model class doesn't allow
+# saving, so saving fails before the router even comes into play. I
+# couldn't figure out a circumstance in a realistic setting where the
+# save method on the model would be bypassed, at least not one that's
+# easily testable.
+@pytest.mark.django_db
```

Comment:
One thing you _could_ do if you wanted is to just mark the entire module as having access to the db, since each test here uses it. Again, don't have to, but if you want to, it saves a couple lines. Would just need to add
```
pytestmark = pytest.mark.django_db
```
after the imports